dotCMS/src/main/java/com/dotcms/rest/TagResource.java

```diff
@@ -44,16 +52,52 @@ protected TagResource(TagAPI tagAPI, WebResource webResource) {
     @JSONP
     @NoCache
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-    public Map<String, RestTag> list(@Context HttpServletRequest request) {
-        TagTransform transform = new TagTransform();
-        List<Tag> tags = getTagsInternal();
+    public Map<String, RestTag> list(@Context final HttpServletRequest request,
+            @QueryParam("name") final String  tagName,
+            @QueryParam("siteOrFolder") final String siteOrFolderId) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+
+        final User user = initDataObject.getUser();
+
+        List<Tag> tags;
+
+        if(UtilMethods.isSet(tagName)) {
+            tags = searchTagsInternal(tagName, siteOrFolderId, user);
+        } else {
+            tags = getTagsInternal();
+        }
+
         Map<String, RestTag> hash = Maps.newHashMapWithExpectedSize(tags.size());
+        TagTransform transform = new TagTransform();
         for (Tag tag : tags) {
             hash.put(tag.getTagName(), transform.appToRest(tag));
         }
         return hash;
```

Comment:
this is not following the standard response used for the endpoint ResponseEntityView

Reply:
Talked to will about it and he doesn't want to change which means creating a new version of this

Reply:
Already discussed @jdotcms